mr/operator_executor.h

```diff
@@ -68,11 +85,13 @@ class OperatorExecutor : public std::enable_shared_from_this<OperatorExecutor> {
 
   /// I keep it as std::map to print counters in lexicographic order.
   /// Performance is negligible since it's used only for final aggregation.
-  std::map<std::string, long> metric_map_;
+  MetricMap metric_map_;
   std::atomic<uint64_t> parse_errors_{0};
 
   RawContext::FreqMapRegistry freq_maps_;
   const RawContext::FreqMapRegistry* finalized_maps_;
+
+  static thread_local std::unique_ptr<PerIoStruct> per_io_;
```

Comment:
One, very strong reason why several concurrent pipelines can not run is because of those thread-local structs. if 2 pipelines run concurrently they will share per_io_ struct_ which will cause data corruption.

Reply:
Shit, you are correct, I didn't think about it. Although maybe the problem is that we are relying on thread_local instead of providing each thread with a pointer to a `new`ly allocated data structure.

Scheiße, du hast rechts, ich habe daran nicht gedacht. Vielleicht ist das Problem, dass wir thread_local benutzen. Stattdessen sollen wir eine `neu`e Datastruktur pro Thread erstellen, was diese Datakorruption verhindern wird.